index.md

```diff
@@ -765,50 +768,56 @@ class Composition(object):
 
 ### Different method signature (W0222) {#W0222}
 
-When declaring a method in a child class, if you're going to provide a
-method name that is the same as a method in the parent class, both the
-parent and child should not differ in their signature (the number of
-arguments).
-
-When you have a method with the same name, the arguments should stay the
-same.
+When a child class overrides a method of the parent class, the new method should have the same signature as the method which it is overriding. In other words, the names and the order of the parameters should be the same in the two methods.
 
 ~~~~ {include="W0222_signature_differs"}
 ~~~~
 
 
 ### Return in `__init__` (E0101) {#E0101}
 
-This error occurs when a return statement is used in the `__init__` method.
-The purpose of this method is only to initialize the attributes of an object, and
-it does not return anything directly.
+This error occurs when the [`__init__`][object.__init__] method contains a return statement.
+
+The purpose of the `__init__` method is to initialize the attributes of an object. `__init__` is called by the special method [`__new__`][object.__new__] when a new object is being instantiated, and `__new__` will raise a `TypeError` if `__init__` returns anything other than `None`.
 
 ~~~~ {include="E0101_return_in_init"}
 ~~~~
 
-
 ### Protected member access (W0212) {#W0212}
 
-Variable names starting with underscores are a convention that means the field should
-not be accessed outside of the calling class. This encapsulation is a
-hint to the user that they should not change the field as it may be
-critical to the proper functioning of the object. Furthermore, this also
-applies to methods with underscores since calling them may also cause
-adverse affects.
+Variable names starting with underscores are a convention that means the field should not be accessed outside of the calling class. This encapsulation is a hint to the user that they should not change the field, as it may be critical to the proper functioning of the object. Furthermore, this also applies to methods with underscores, since calling them may also cause adverse effects.
```

Comment:
Also, change last part to "this also applies to methods *whose names start* with underscores".